### PR TITLE
feat(ingestion/airflow): Added support of datajob entity as input lineage

### PR DESCRIPTION
**Pull request by shubhamjagtap639**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py [hunk 1]
```diff
@@ -17,9 +17,9 @@
 assert AIRFLOW_PATCHED
 
 if TYPE_CHECKING:
-    from airflow import DAG
     from airflow.models import DagRun, TaskInstance
 
+    from airflow import DAG
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py]: why'd this change?

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py [hunk 2]
```diff
@@ -272,10 +275,9 @@ def _extract_lineage(
         )
 
         # Write the lineage to the datajob object.
-        datajob.inlets.extend(DatasetUrn.create_from_string(urn) for urn in input_urns)
-        datajob.outlets.extend(
-            DatasetUrn.create_from_string(urn) for urn in output_urns
-        )
+        datajob.inlets.extend(entities_to_dataset_urn_list(input_urns))
```
Comment (hsheth2) [inline metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py:278]: how about a `entities_to_urn_list(list, allowed_entities={"dataset, "datajob"})`

**Review by hsheth2 — COMMENTED** [review]
one suggestion, mostly looking good

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py (hunk 2 above)
Comment (shubhamjagtap639) [reply metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/datahub_listener.py:278]: If we use common method to extend both iolets and upstrem_jobs, we get lint error as:
`src/datahub_airflow_plugin/datahub_listener.py:279: error: Argument 1 to "extend" of "list" has incompatible type "list[DatasetUrn | DataJobUrn]"; expected "Iterable[DatasetUrn]"  [arg-type]`
with below method definition:
`def entities_to_urn_list(
    iolets: List[str], allowed_entities: Set[str]
) -> List[Union[DatasetUrn, DataJobUrn]]:
    urn_list: List[Union[DatasetUrn, DataJobUrn]] = []
    for let in iolets:
        entity = guess_entity_type(let)
        if entity in allowed_entities:
            if entity == "dataset":
                urn_list.append(DatasetUrn.create_from_string(let))
            elif entity == "dataJob":
                urn_list.append(DataJobUrn.create_from_string(let))
    return urn_list`

File: metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py (hunk 1 above)
Comment (shubhamjagtap639) [reply metadata-ingestion-modules/airflow-plugin/src/datahub_airflow_plugin/client/airflow_generator.py:22]: This changes happens after running lint fix command:
./gradlew :metadata-ingestion-modules:airflow-plugin:lintFix
and Github CL is getting failed due to this changes.
Can you tell me why same command is behaving different on local and github cl?

**Comment by hsheth2** [comment]
@shubhamjagtap639 can you move this PR to open-source?

**Pull request closed without merging**